Branch-SDK/src/io/branch/referral/util/BranchEvent.java

```diff
@@ -1,21 +1,303 @@
+        this.eventName = eventName;
+        this.isStandardEvent = isStandardEvent;
+        buoList = new ArrayList<>();
+    }
+
+    /**
+     * Set the transaction id associated with this event if there in any
+     *
+     * @param transactionID {@link String transactionID}
+     */
+    public BranchEvent setTransactionID(String transactionID) {
+        return addStandardProperty(Defines.Jsonkey.TransactionID.getKey(), transactionID);
+    }
+
+    /**
+     * Set the currency related with this transaction event
+     *
+     * @param currency iso4217Code for currency. Defined  in {@link CurrencyType}
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setCurrency(CurrencyType currency) {
+        return addStandardProperty(Defines.Jsonkey.Currency.getKey(), currency.toString());
+    }
+
+    /**
+     * Set the revenue value  related with this transaction event
+     *
+     * @param revenue {@link double } revenue value
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setRevenue(double revenue) {
+        return addStandardProperty(Defines.Jsonkey.Revenue.getKey(), revenue);
+    }
+
+    /**
+     * Set the shipping value  related with this transaction event
+     *
+     * @param shipping {@link double } shipping value
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setShipping(double shipping) {
+        return addStandardProperty(Defines.Jsonkey.Shipping.getKey(), shipping);
+    }
+
+    /**
+     * Set the tax value  related with this transaction event
+     *
+     * @param tax {@link double } tax value
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setTax(double tax) {
+        return addStandardProperty(Defines.Jsonkey.Tax.getKey(), tax);
+    }
+
+    /**
+     * Set any coupons associated with this transaction event
+     *
+     * @param coupon {@link String } with any coupon value
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setCoupon(String coupon) {
+        return addStandardProperty(Defines.Jsonkey.Coupon.getKey(), coupon);
+    }
+
+    /**
+     * Set any affiliation for this transaction event
+     *
+     * @param affiliation {@link String } any affiliation value
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setAffiliation(String affiliation) {
+        return addStandardProperty(Defines.Jsonkey.Affiliation.getKey(), affiliation);
+    }
+
+    /**
+     * Set the condition of the content item. Value is one of the enum constants from {@link BranchEvent.CONDITION}
+     *
+     * @param condition {@link BranchEvent.CONDITION}
+     * @return This object for chaining builder methods
+     */
+    public BranchEvent setCondition(CONDITION condition) {
+        return addStandardProperty(Defines.Jsonkey.Condition.getKey(), condition.name());
+    }
```

Comment:
this belongs on contentmetadata, not on the branchevent

Reply:
Agree that it make sense to have it in ContentMetadata. Is our back end is ready to handle it if we move this to ContentMetadata